tests/core_tests/test_api.py

```diff
@@ -210,6 +210,17 @@ def test_skip_invalid(self):
         self.assertIn(good_file_path, result)
         self.assertNotIn(bad_file_path, result)
 
+    def test_filenames_without_optional_keys(self):
+        template = self.tk.templates["path_with_optional_abstract"]
```

Comment:
please provide a description of the tests and some comments

Reply:
I noticed you added comments elsewhere (thank you!), but this file still needs some.